Sources/Moya/JSONDecoder+KeyPath.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+internal extension JSONDecoder {
+
+    /// Decode value at the keypath of the given type from the given JSON representation
+    ///
+    /// - Parameters:
+    ///   - type: The type of the value to decode.
+    ///   - data: The data to decode from.
+    ///   - keyPath: The JSON keypath
+    ///   - keyPathSeparator: Nested keypath separator
+    /// - Returns: A value of the requested type.
+    /// - Throws: An error if any value throws an error during decoding.
+    func decode<T>(_ type: T.Type,
+                   from data: Data,
+                   keyPath: String,
+                   keyPathSeparator separator: String = ".") throws -> T where T: Decodable {
+        userInfo[keyPathUserInfoKey] = keyPath.components(separatedBy: separator)
+        return try decode(KeyPathWrapper<T>.self, from: data).object
+    }
+}
+
+/// The keypath key in the `userInfo`
+private let keyPathUserInfoKey = CodingUserInfoKey(rawValue: "keyPathUserInfoKey")! //swiftlint:disable:this force_unwrapping
+
+/// Object which is representing the value
+private final class KeyPathWrapper<T: Decodable>: Decodable {
+
+    enum KeyPathError: Error {
+        case `internal`
+    }
+
+    /// Naive coding key implementation
+    struct Key: CodingKey {
+        init?(intValue: Int) {
```

Comment:
Why do we need a key for integer?

Reply:
It is requirements of the `CodingKey` protocol:
> The value to use in an integer-indexed collection (e.g. an int-keyed dictionary).
**Required.**